File: HomeWorks/src/Lesson6/Hands/SamsungHand.java
package Lesson6.Hands;

public class SamsungHand implements IHand {
    private int price;

    public SamsungHand() {}

    public SamsungHand(int price) {
        this.price = price;
    }

    @Override
    public void upHand() {
        System.out.println("Машет рука Samsung");
    }


    public void setPrice(int price) {
        this.price = price;
    }

    @Override
    public int getPrice() {
        return price;
    }
}

File: HomeWorkKotlin/src/Robot/Hands/SamsungHand.kt
package Robot.Hands

class SamsungHand(private val priceParam : Int) : IHand {

    override var price = priceParam
        set(value) {
            if(value > 0) field = value
        }

    override fun upHand() {
        println("Машет рука Samsung")
    }
}